src/main/java/ch/njol/skript/expressions/ExprNumbers.java

```diff
@@ -81,10 +81,14 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 			s = f;
 			f = temp;
 		}
-		final double amount = integer ? Math.floor(f.doubleValue()) - Math.ceil(s.doubleValue()) + 1 : Math.floor(f.doubleValue() - s.doubleValue() + 1);
 		final List<Number> list = new ArrayList<>();
 		final double low = integer ? Math.ceil(s.doubleValue()) : s.doubleValue();
-		for (int i = 0; i < amount; i++) {
+		double decimalCheck = s.doubleValue() % 1;

```

Comment:
This doesn't account for floating-point precision errors, see [this repl.](https://repl.it/@blightedblu/Demonstration-FPE-with-n-1-in-Java)

Reply:
Oh dear I missed that, yeah use ``s == Math.rint(s)`` as the check comparison instead, with ``decimalCheck`` being boolean instead of double for this.